lib/wasi/src/runtime/mod.rs

```diff
@@ -26,24 +32,11 @@ where
     /// Retrieve the active [`VirtualTaskManager`].
     fn task_manager(&self) -> &Arc<dyn VirtualTaskManager>;
 
-    /// Get a [`wasmer::Engine`] for module compilation.
-    fn engine(&self) -> Option<wasmer::Engine> {
-        None
-    }
+    fn package_resolver(&self) -> Arc<dyn PackageResolver + Send + Sync>;
 
     /// Create a new [`wasmer::Store`].
     fn new_store(&self) -> wasmer::Store {
-        cfg_if::cfg_if! {
-            if #[cfg(feature = "sys")] {
-                if let Some(engine) = self.engine() {
```

Comment:
Sorry, I saw the `engine()` method was unused throughout the repo except for our default `new_store()` implementation and one line in the `ModuleCache` that I have since removed, so I thought it was dead code.

I've put it back in.